flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -414,12 +427,18 @@ private void genAgentData(ImplicitNewExpressionNode newExpressionNode, ClassSymb
             List<ToolData> toolsData = new ArrayList<>();
             ListConstructorExpressionNode listCtrExprNode = (ListConstructorExpressionNode) toolsArg;
             for (Node node : listCtrExprNode.expressions()) {
+                if (node.kind() == SyntaxKind.CHECK_EXPRESSION) {
+                    String iconPath = "https://bcentral-packageicons.azureedge.net/images/ballerina_mcp_0.4.2.png";
```

Comment:
Move this variable to the static-level, with the expression `CommonUtils.generateIcon("ballerina", "mcp", "0.4.2")`. On a separate note, it would be better if we could derive the MCP version from the user's code. Since this isn't necessary for this PR, let's add add a comment stating this.